### DISCUSSION TITLE
__VUE_PROD_DEVTOOLS__ Cause my project to not run.

### DISCUSSION THREAD

**Discussion by xstxhjh** [vitejs/vite-plugin-vue#451] [Q&A]
### Describe the bug

vite.config.ts
define: {
  __VUE_PROD_DEVTOOLS__: JSON.stringify(true),
},

If I change VUE_PROD_DEVTOOLS to false or delete this line of code, it will cause the browser of a certain page in the project to freeze directly.

### Reproduction

https://ali.dev.goviewlink.com/chart/home/clqap150z0001mykzdn1q3mv8

### Steps to reproduce

username: 13476068653
password: a123456
Log in and open the URL
https://ali.dev.goviewlink.com/chart/home/clqap150z0001mykzdn1q3mv8

I've tried to reproduce this bug, but it only appears on a page in my real environment.

### System Info

```shell
"@vitejs/plugin-vue": "^4.5.2"

System:
    OS: macOS 14.1.2
    CPU: (11) arm64 Apple M3 Pro
    Memory: 299.58 MB / 18.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.19.0 - ~/.nvm/versions/node/v18.19.0/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v18.19.0/bin/yarn
    npm: 10.2.3 - ~/.nvm/versions/node/v18.19.0/bin/npm
    bun: 1.0.15 - ~/.bun/bin/bun
  Browsers:
    Safari: 17.1.2
  npmPackages:
    @vitejs/plugin-vue: ^4.5.2 => 4.2.3 
    @vitejs/plugin-vue-jsx: ^3.1.0 => 3.0.1 
    vite: ^5.0.10 => 5.0.10
```

### Used Package Manager

npm

### Logs

Project stuck

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [x] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by xstxhjh** [comment]
Who can provide me with help, I can locate the problem, because really modify the __VUE_PROD_DEVTOOLS__ variable will cause the page to get stuck.

**Comment by xstxhjh** [comment]
https://github.com/vuejs/devtools/issues/2103 
I see a lot of people who have the same problem as me.

**Comment by xstxhjh** [comment]
I don't think this is a bug with the chrome plugin, because I have this problem when opening it in other browsers, and there is no installation tool in my project.

**Comment by xstxhjh** [comment]
<img width="636" alt="image" src="https://github.com/vitejs/vite-plugin-vue/assets/33480698/0c7f99b5-5396-4729-999f-d04a4bb4821e">

Why __VUE_PROD_DEVTOOLS__ be affected by this logic and cause the project to freeze.

**Comment by edison1105** [comment]
try https://github.com/vuejs/devtools-next
